mobly/utils.py

```diff
@@ -603,13 +603,22 @@ def wait_for_standing_subprocess(proc, timeout=None):
 def get_available_host_port():
   """Gets a host port number available for adb forward.
 
+  DEPRECATED: This method is inherently racy, instead you should pass `tcp:0` to
+  adb and let adb take care of the port allocation.
+
   Returns:
     An integer representing a port number on the host available for adb
     forward.
 
   Raises:
     Error: when no port is found after MAX_PORT_ALLOCATION_RETRY times.
   """
+  logging.warning(
+      'The method mobly.utils.get_available_host_port is deprecated because it '
+      'is inherently racy. Instead you should pass "tcp:0" to adb and let adb '
```

Comment:
just say this method is unreliable.

no need to say "racy"

Reply:
Modified